src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -361,6 +362,19 @@ void WbSupervisorUtilities::reset() {
   initControllerRequests();
 }
 
+const WbNode *WbSupervisorUtilities::getNodeFromProtoDEF(const WbNode *fromNode, const QString &defName) const {
+  // recursively search in PROTO body for the DEF node
+  QList<WbNode *> descendants = fromNode->subNodes(false, true, false);  // get nodes from PROTO fields
```

Comment:
I slightly changed the search behavior for `wb_supervisor_node_get_from_def`:
* previously: the search was made recursively in fields only:
    i.e. also in case of nested PROTO fields were searched instead of parameters
* now: for the initial PROTO node we search in fields but for all the other nested PROTO nodes we search in parameters.
    This would be like opening a PROTO file and searching for a DEF name in the body without searching in other files.